compiler-cli/src/dependencies.rs

```diff
@@ -201,30 +203,52 @@ pub fn download<Telem: Telemetry>(
     Ok(manifest)
 }
 
-async fn download_missing_packages<Telem: Telemetry>(
-    downloader: hex::Downloader,
+async fn add_missing_packages<Telem: Telemetry>(
+    paths: &ProjectPaths,
+    fs: Box<ProjectIO>,
     manifest: &Manifest,
     local: &LocalPackages,
     project_name: SmolStr,
     telemetry: &Telem,
 ) -> Result<(), Error> {
-    let mut count = 0;
-    let mut missing = local
-        .missing_local_packages(manifest, &project_name)
+    let missing_packages = local.missing_local_packages(manifest, &project_name);
+
+    // Link local paths
+    let packages_dir = paths.build_packages_directory();
+    for package in missing_packages.iter() {
+        let package_dest = packages_dir.join(package.name.to_string());
+        match &package.source {
+            ManifestPackageSource::Hex { .. } => Ok(()),
+            ManifestPackageSource::Local { path } => fs.symlink_dir(&path, &package_dest),
```

Comment:
Symlinks are not supported on Windows so we'll need to do something else here.

Long term this is likely using the build directory within the target directory itself, but for now we could copy the `gleam.toml`, `src`, and `priv`.

Reply:
Is this suggestion intended for all platforms to maintain uniformity, or just for windows?

Reply:
Each approach would have different user facing behaviour when files get edited, so I think we should do the same thing on all platforms.